src/core/sr.ml

```diff
@@ -136,19 +136,27 @@ let symb_to_tenv
    [Fatal] is raised in case of error. *)
 let check_rule : Scope.pre_rule Pos.loc -> rule = fun pr ->
   (* Unwrap the contents of the pre-rule. *)
-  let (pos, s, lhs, vars, rhs_vars, arities) =
-    let Pos.{elt=Scope.{pr_sym;pr_lhs;pr_vars;pr_rhs;pr_arities;_}; pos} = pr
+  let (pos, s, lhs, vars, rhs_vars, arities, xvars) =
+    let Pos.{elt=Scope.{pr_sym;pr_lhs;pr_vars;pr_rhs;pr_arities;
```

Comment:
Perhaps we should use a let here to make things more readable.

Reply:
Agreed.